src/benchmark.ts

```diff
@@ -0,0 +1,115 @@
+import { CommitFlags, CommitTransfer, CreateAccount, createClient, CreateTransfer } from '.'
+
+const REQUEST_BATCH_SIZE = 10000
+const MAX_TRANSFERS = 600000
+
+const client = createClient({
+  cluster_id: 0x0a5ca1ab1ebee11en,
+  replica_addresses: ['3001']
+})
+
+const accountA: CreateAccount = {
+  id: 137n,
+  user_data: 0n,
+  reserved: Buffer.alloc(48, 0),
+  code: 0,
+  flags: 0,
+  unit: 1,
+  debits_accepted: 0n,
+  debits_reserved: 0n,
+  credits_accepted: 0n,
+  credits_reserved: 0n,
+  timestamp: 0n,
+}
+
+const accountB: CreateAccount = {
+  id: 138n,
+  user_data: 0n,
+  reserved: Buffer.alloc(48, 0),
+  code: 0,
+  flags: 0,
+  unit: 1,
+  debits_accepted: 0n,
+  debits_reserved: 0n,
+  credits_accepted: 0n,
+  credits_reserved: 0n,
+  timestamp: 0n,
+}
+
+// pre-allocate 1 million transfers and commits
+console.log(`pre-allocating ${MAX_TRANSFERS} transfers and commits...`)
+const transfers: CreateTransfer[] = []
+const commits: CommitTransfer[] = []
+
+for (let i = 0; i < MAX_TRANSFERS; i++) {
+  transfers.push({
+    id: BigInt(i),
+    debit_account_id: accountA.id,
+    credit_account_id: accountB.id,
+    code: 0,
+    reserved: Buffer.alloc(32, 0),
```

Comment:
Sharing a single pre-allocated Reserved32 across all these transfers and commits will reduce the memory requirements of the benchmark by a few hundred megs. It will also speed up the benchmark init routine, less allocations, less zeroing of allocated buffers.

Reply:
@wilsonianb just a heads up on our Node client benchmark, it's not at full throttle yet because we only tick the client's IO (to see if any IO jobs have finished) once every 15ms at present. We have plans to optimize this after BetaBeetle, to integrate better with Node's event loop, but it should at least already be faster than Node's max HTTP requests per second surface ceiling.